core/src/main/java/com/datastrato/gravitino/storage/relational/service/SchemaMetaService.java

```diff
@@ -0,0 +1,216 @@
+import com.datastrato.gravitino.exceptions.NoSuchEntityException;
+import com.datastrato.gravitino.meta.SchemaEntity;
+import com.datastrato.gravitino.storage.relational.mapper.SchemaMetaMapper;
+import com.datastrato.gravitino.storage.relational.po.SchemaPO;
+import com.datastrato.gravitino.storage.relational.utils.POConverters;
+import com.datastrato.gravitino.storage.relational.utils.SessionUtils;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+/** The service class for schema metadata. It provides the basic database operations for schema. */
+public class SchemaMetaService {
+  private static final SchemaMetaService INSTANCE = new SchemaMetaService();
+
+  public static SchemaMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  private SchemaMetaService() {}
+
+  public SchemaPO getSchemaPOByCatalogIdAndName(Long catalogId, String schemaName) {
+    SchemaPO schemaPO =
+        SessionUtils.getWithoutCommit(
+            SchemaMetaMapper.class,
+            mapper -> mapper.selectSchemaMetaByCatalogIdAndName(catalogId, schemaName));
+
+    if (schemaPO == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE,
+          Entity.EntityType.SCHEMA.name().toLowerCase(),
+          schemaName);
+    }
+    return schemaPO;
+  }
+
+  public Long getSchemaIdByCatalogIdAndName(Long catalogId, String schemaName) {
+    Long schemaId =
+        SessionUtils.getWithoutCommit(
+            SchemaMetaMapper.class,
+            mapper -> mapper.selectSchemaIdByCatalogIdAndName(catalogId, schemaName));
+
+    if (schemaId == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE,
+          Entity.EntityType.SCHEMA.name().toLowerCase(),
+          schemaName);
+    }
+    return schemaId;
+  }
+
+  public SchemaEntity getSchemaByIdentifier(NameIdentifier identifier) {
+    NameIdentifier.checkSchema(identifier);
+    String metalakeName = identifier.namespace().level(0);
+    String catalogName = identifier.namespace().level(1);
+    String schemaName = identifier.name();
+
+    Long metalakeId = MetalakeMetaService.getInstance().getMetalakeIdByName(metalakeName);
+
+    Long catalogId =
+        CatalogMetaService.getInstance().getCatalogIdByMetalakeIdAndName(metalakeId, catalogName);
+
+    SchemaPO schemaPO = getSchemaPOByCatalogIdAndName(catalogId, schemaName);
+
+    return POConverters.fromSchemaPO(schemaPO, identifier.namespace());
+  }
+
+  public List<SchemaEntity> listSchemasByNamespace(Namespace namespace) {
+    Namespace.checkSchema(namespace);
+    String metalakeName = namespace.level(0);
+    String catalogName = namespace.level(1);
+
+    Long metalakeId = MetalakeMetaService.getInstance().getMetalakeIdByName(metalakeName);
```

Comment:
Could you extract a common method for schema, catalog and metalake to get entity id to avoid duplicated code?
```
Long getMetaEntityId(Namespace/ NameIdentifier id) {
    for (int level : namepace.levels()) {
         switch (level): {
             case 0: id = getMetalake();
             case 1: id = getCatalog();
         }
    }
}
```

Reply:
Consider adding a `CommonMetaService` class to handle the situation you mentioned.

Reply:
For now, maybe a common method is enough good. But maybe it will be better if we have a `CommonMetaService`.

Reply:
Could you set the SchemaPoBuilder in this method to address this issue?

Reply:
The logic of `insert` and `update` when going to the underlying resource is quite different depending on whether it has a `version` table. Adding `POBuilder` logic to this method may cause it to become complicated. My idea is to keep the logic of `insert` and `update` for the time being, maybe it's better to consider this after we merging all the operation PRs.

Reply:
cc @yuqi1129 WDYT?

Reply:
@xloya You don't get my point.
You can add a new my method
```
void schemaBuilderIdentifierField(SchemaPoBuilder builder,  Namespace namespace) {
          for (int level : namepace.levels()) {
         switch (level): {
             case 0: id = getMetalake();
                    builder.setMetalakeId();
             case 1: id = getCatalog();
                   builder.setCatalogId();
         }
    }
}
```

Reply:
Done